packages/reaction-core/server/import.js

```diff
@@ -315,6 +315,79 @@ ReactionImport.tag = function (key, tag) {
 };
 
 /**
+ * @summary Returns an disjoint object as right join. For a visualization, see:
+ *          http://www.codeproject.com/KB/database/Visual_SQL_Joins/Visual_SQL_JOINS_orig.jpg
+ *          Additionally, the join is done recursively on properties of
+ *          nested objects as well. Nested arrays are handled like
+ *          primitive values.
+ * @param {Object} leftSet An object that can contain nested sub-objects
+ * @param {Object} rightSet An object that can contain nested sub-objects
+ * @returns {Object} The disjoint object that does only contain properties
+ *                   from the rightSet. But only those, that were not present
+ *                   in the leftSet.
+ */
+function doRightJoinNoIntersection (leftSet, rightSet) {
+  if (rightSet === null) return null;
+
+  let rightJoin;
+  if (Array.isArray(rightSet)) {
+     rightJoin = [];
+  } else {
+     rightJoin = {};
+  }
+  let findRightOnlyProperties = function () {
+    return Object.keys(rightSet).filter(function(key) {
+      if (typeof(rightSet[key]) === "object" &&
+          !Array.isArray(rightSet[key])) {
+            // Nested objects are always considered
+            return true;
+      } else {
+        // Array or primitive value
+        return !leftSet.hasOwnProperty(key);
+      }
+    })
+  };
+
+  for (let key of findRightOnlyProperties()){
+    if (typeof(rightSet[key]) === "object") {
+      // subobject or array
+      if (leftSet.hasOwnProperty(key) && (typeof(leftSet[key]) !== "object" ||
+           Array.isArray(leftSet[key])!== Array.isArray(rightSet[key]))) {
+        // This is not expected!
+        throw new Error(
```

Comment:
We should probably add a bunyan log here as well. just for best practices...  could also use Meteor.error